synapse/lib/layer.py

```diff
@@ -1204,6 +1213,182 @@ async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, map
 
         self.onfini(self._onLayrFini)
 
+    async def verifyBuidTag(self, buid, formname, tagname, tagvalu):
+        abrv = self.tagabrv.bytsToAbrv(tagname.encode())
+        abrv += self.getPropAbrv(formname, None)
+        if not self.layrslab.hasdup(abrv, buid, db=self.bytag):
+            yield ('NoTagIndex', {'buid': buid, 'form': formname, 'tag': tagname, 'valu': tagvalu})
+
+    def _testDelTagIndx(self, buid, form, tag):
```

Comment:
Should these test methods call layr sync so we ensure they are commited txns?

Reply:
sure :)

Reply:
Actually, if you mean commit() it's def not necessary